lib/image.py

```diff
@@ -1,30 +1,72 @@
 from uuid import uuid4
-from PIL import Image
+from PIL import Image, ImageFile
 from io import BytesIO
 from django.core.files.uploadedfile import SimpleUploadedFile
+from django.utils.deconstruct import deconstructible
 import os
 
 
-# Intended to be used as an argument for upload_to option in ImageField
-# Generates a random path for an image formatted as
-# 'MEDIA_ROOT/<args[0]>/.../<random id>.<ext>'
+# Hack for making pillow accept large images
+# (should maybe be removed after importing images)
+ImageFile.SAFEBLOCK = 1024*10000
+
+
+def path():
+    pass
+
+
+# Check if a thumbnail should be generated or not
+# Based on original image uploaded with uuid
+# and generated image having the same name
+def should_generate(original, generated):
+    if original:
+        if generated is None:
+            return True
+        else:
+            original_name, _ = os.path.splitext(original.name)
```

Comment:
This should probably be:
original_name, _ = os.path.splitext(os.path.basename(original.name))